### ISSUE TITLE
Namespace getters case

### ISSUE THREAD

**Issue by guybedford** [rollup/rollup#3923]
### Feature Use Case

Just saw this discussed on esbuild here - https://github.com/evanw/esbuild

https://rollupjs.org/repl/?version=2.36.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMG1hdGhzJTIwZnJvbSUyMCcuJTJGbWF0aHMuanMnJTNCJTVDbmxldCUyMGtleSUyMCUzRCUyMCdzcXVhcmUnJTNCJTVDbmNvbnNvbGUubG9nKG1hdGhzKSUzQiU1Q25jb25zb2xlLmxvZyhtYXRocyU1QmtleSU1RCklM0IlNUNubWF0aHMuc3F1YXJlKGZ1bmN0aW9uJTIwb3RoZXIoKSUyMCU3QiU3RCklM0IlNUNuY29uc29sZS5sb2cobWF0aHMlNUJrZXklNUQpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybWF0aHMuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwc3F1YXJlJTIwZnJvbSUyMCcuJTJGc3F1YXJlLmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwJTdCZGVmYXVsdCUyMGFzJTIwc3F1YXJlJTdEJTIwZnJvbSUyMCcuJTJGc3F1YXJlLmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwbGV0JTIwdmFsdWUlMjAlM0QlMjAwJTNCJTVDbiU1Q25leHBvcnQlMjBmdW5jdGlvbiUyMGN1YmUlMjAoeCUyMCklMjAlN0IlNUNuJTVDdHJldHVybiUyMHNxdWFyZSh4KSUyMColMjB4JTNCJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJzcXVhcmUuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZGVmYXVsdCUyMGZ1bmN0aW9uJTIwc3F1YXJlJTIwKCUyMHglMjApJTIwJTdCJTVDbiU1Q3RzcXVhcmUlMjAlM0QlMjB4JTNCJTVDbiU1Q3RyZXR1cm4lMjB4JTIwKiUyMHglM0IlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

### Feature Proposal

We should use getters on created namespaces whenever there exist known assignments to exported bindings.

**Comment by lukastaegert** [comment]
Not sure I understand your example but we are using getters for known reassignments: https://rollupjs.org/repl/?version=2.36.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMG1hdGhzJTIwZnJvbSUyMCcuJTJGbWF0aHMuanMnJTNCJTVDbiU1Q25jb25zb2xlLmxvZyhtYXRocyklM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtYXRocy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBsZXQlMjB2YWx1ZSUyMCUzRCUyMDAlM0IlNUNudmFsdWUlMjAlM0QlMjAxJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

Did we miss some cases?

**Comment by guybedford** [comment]
Yes exactly - it seems this is specifically for the default export function binding update.

That is:

```js
export function q () {
}
q = 5;
export default function p () {
}
p = 10;
```

are both live bindings updates that are valid :)

**Comment by lukastaegert** [comment]
Yes, live-bindings for default exports are an exception. Maybe change the issue accordingly?

**Comment by Chijindui** [comment]
> Not sure I understand your example but we are using getters for known reassignments: https://rollupjs.org/repl/?version=2.36.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHN0dWZmJTIwZnJvbSUyMCcuJTJGc3R1ZmYuanMnJTNCJTVDbiU1Q25jb25zb2xlLmxvZyhzdHVmZiklM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJzdHVmZi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBsZXQlMjBtb2RpZmllZCUyMCUzRCUyMCdub3QlMjBtb2RpZmllZCclM0IlNUNuc2V0VGltZW91dCgoKSUyMCUzRCUzRSUyMG1vZGlmaWVkJTIwJTNEJTIwJ21vZGlmaWVkJyklM0IlNUNuZXhwb3J0JTIwY29uc3QlMjBjb25zdGFudCUyMCUzRCUyMCdjb25zdGFudCclM0IlNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1
